### PR TITLE
Updated buildpackdeps-stretch to latest GitHub runners's cached layers

### PR DESCRIPTION
**Pull request by kichalla**

### CODE REVIEW

File: images/build/Dockerfiles/azureFunctions.JamStack.Dockerfile [hunk 1]
```diff
@@ -5,7 +5,7 @@ ARG BUILD_DIR=/tmp/oryx/build
 # Determine where the image is getting built (DevOps agents or local)
 ARG AGENTBUILD
 
-FROM buildpack-deps:stretch@sha256:8bcd320ec29cf67052985f28891586fb853051f69ad0646fc7a49f47d6e3ee1a AS main
+FROM debian:stretch AS main
```
Comment (arroyc) [inline images/build/Dockerfiles/azureFunctions.JamStack.Dockerfile]: nit: after the discussion, may be it will be better if we add some note that Debian:stretch is as per functions teams current requirement, we may go back to buildpack-deps:stretch in future

File: images/build/Dockerfiles/gitHubActions.Dockerfile [hunk 2]
```diff
@@ -7,9 +7,7 @@ ARG SDK_STORAGE_BASE_URL_VALUE
 # Determine where the image is getting built (DevOps agents or local)
 ARG AGENTBUILD
 
-# Use the exact layer that is present on the GitHub agent machines
-# https://github.com/actions/virtual-environments/blob/master/images/linux/Ubuntu1804-README.md
-FROM buildpack-deps:stretch@sha256:8bcd320ec29cf67052985f28891586fb853051f69ad0646fc7a49f47d6e3ee1a AS main
+FROM githubactions-buildpackdeps-stretch AS main
```
Comment (arroyc) [inline images/build/Dockerfiles/gitHubActions.Dockerfile]: curious: why this change? should it be githubrunner-buildpackdeps-stretch?

File: images/build/Dockerfiles/gitHubRunners.BuildPackDepsStretch.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# The digest here must match with what is present in the following location
+# so that downloading of the image is very fast in GitHub workflows
+# https://github.com/actions/virtual-environments/blob/master/images/linux/Ubuntu1804-README.md
+FROM buildpack-deps:stretch@sha256:dc901bbf4b34e4ca8771c0d0773e557221452f97bcf0c732de7ecda3782bdf97 AS main
```
Comment (arroyc) [inline images/build/Dockerfiles/gitHubRunners.BuildPackDepsStretch.Dockerfile]: what is the use of this dockerfile?

File: images/build/Dockerfiles/azureFunctions.JamStack.Dockerfile (hunk 1 above)
Comment (kichalla) [reply images/build/Dockerfiles/azureFunctions.JamStack.Dockerfile]: Sure, I added a comment

File: images/build/Dockerfiles/gitHubRunners.BuildPackDepsStretch.Dockerfile (hunk 3 above)
Comment (kichalla) [reply images/build/Dockerfiles/gitHubRunners.BuildPackDepsStretch.Dockerfile]: To have one central place for generating a docker image based on the Github runner layers. In future if AzFunctions wants to use the same layers, we can just use this image instead.

File: images/build/Dockerfiles/gitHubActions.Dockerfile (hunk 2 above)
Comment (kichalla) [reply images/build/Dockerfiles/gitHubActions.Dockerfile]: check my below comment.
